backend/shared/removeConnection.ts

```diff
@@ -1,14 +1,19 @@
+import { broadcastState } from './broadcast-state';
 import { deleteConnection } from './database/deleteConnection';
 import { getConnection } from './database/getConnection';
 import { removeConnectionFromGroup } from './database/removeConnectionFromGroup';
-import { Config } from './types';
+import { ConfigWithHandler } from './types';
 
-export const removeConnection = async (config: Config): Promise<unknown> => {
+export const removeConnection = async (config: ConfigWithHandler): Promise<unknown> => {
   const connectionItem = await getConnection(config);
   if (!connectionItem) return;
+
   const { groupId, userId } = connectionItem;
-  return Promise.all([
-    deleteConnection(config),
+  const [groupItem] = await Promise.all([
     userId && groupId && removeConnectionFromGroup(userId, groupId, config),
+    deleteConnection(config),
```

Comment:
I just noticed that this PR introduced a import cycle that vite warns about: `removeConnection -> broadcastState -> sendMessageToConnection -> removeConnection`. However, functionality seems unaffected.

@lukastaegert Is this acceptable? If not, do you have a suggestion how to resolve it? A trivial solution would be to bundle all of those functions into a single file.

Reply:
Yes, that is indeed not nice. While it may work, there is always a danger that future code changes introduce problem when there is a cyclic dependency, or that some problems only surface after bundling. Admittedly, I also cannot find a better solution than to move them into a single file. Sometimes, such issues can be solved via dependency injection, i.e. hand one of the dependencies over as a call parameter to the next one, but I do not see how to apply such a pattern here in a nice way.